src/parser/smt2/smt2.cpp

```diff
@@ -1138,10 +1103,31 @@ api::Term Smt2::applyParseOp(ParseOp& p, std::vector<api::Term>& args)
     }
     else if (kind == api::MINUS && args.size() == 1)
     {
+      if (isConstInt(args[0]) && args[0].getRealOrIntegerValueSign() > 0)
```

Comment:
So `(- 0)` would still yield `(UMINUS 0)`? Can the sign actually be negative here?

Reply:
Yes and yes.
(- 0) would be (UMINUS CONST_RATIONAL{0}) and (- (- 6)) would be (UMINUS CONST_RATIONAL{-6}).  This is expected since these are not SMT-LIB syntax for values.